### DISCUSSION TITLE
Add "since vX" to rule set docs?

### DISCUSSION THREAD

**Discussion by bsara** [detekt/detekt#3460] [Ideas]
It would be nice to know when rules (and accompanying configuration options) were introduced into Detekt. That way you know whether or not the version that you're currently using actually supports the rule/config option. What do you think?

Examples:

---

![image](https://user-images.githubusercontent.com/3743423/107258906-9905cf00-69f9-11eb-8c15-ec534763d792.png)

---

![image](https://user-images.githubusercontent.com/3743423/107259179-ed10b380-69f9-11eb-8356-eca1db2c570d.png)

**Comment by BraisGabin** [comment]
I like the idea!

**Comment by cortinico** [comment]
That's great, would you like to contribute and do it? 🙏

**Reply by bsara** [reply]
I would love to...I'm just pretty busy with some things and don't have the time at the moment. :(

On a note similar to this, I've noticed that it is difficult to determine exactly all of the new rules and new/changed options for rules in the changelogs. Perhaps having specific sections in future changelogs would help? Something like the following maybe?

```
### New Rules
- ...
- ...

### Updated Rules
- ...
- ...

### Removed Rules
- ...
```

**Reply by cortinico** [reply]
>  Perhaps having specific sections in future changelogs would help? Something like the following maybe?

That's great input. I've worked a bit on our changelog script. I will work on adding the sections you mentioned.

**Reply by bsara** [reply]
thanks so much!

**Comment by cortinico** [comment]
As a small side note: we now show on the website the version when a rule was **enabled** (not added, we don't have that information from the code easily).
